File: homework_1/seven_irem.rb
# foo function to print pairs
def foo(k, temp)
    print "Key is: ", k, " Value is: ", temp[k]
    puts
  end

#initialize hash
arr = { "Name" => "Jane", "City" => "Amsterdam", "Age" => 32, "Weight" => 58}
puts "**Initial Hash**", arr   #printing first version

puts "Value for key Name is ", arr["Name"]   #get the value for Name key
puts "Value for key Salary is ", arr["Salary"] #get the value for nonexistent key

puts ""
arr["Height"] = 162   #add new element

arr.delete("City")   #remove new element
arr.delete("Salary")   #remove nonexistent element

arr["Age"] = 33   #modify Age 

#search for existance of a key
puts "City key existance: ", arr.key?("City")
#search for existance of a value
puts "Value Amsterdam existance: ", arr.has_value?(162)

#print modified version in loop by foo
puts "\n**Modified Hash**"
arr.each do |key, value|
  foo(key, arr)
end

File: homework_1/seven_irem.dart
foo(String k, Map m) {
  print("Key : $k, value : ${m[k]}");
}

void main() {   
  //initialize
  var arr = {'Name':'John', 'City':'Rome', 'Age': 28, 'Height': 182}; 
  print(arr["Name"]); //get the value for a given key
  print(arr["Salary"]); //get the value for nonexistent key
  
  print('**Initial Version**');
  print(arr);   //print initial vesion
  arr['Weight'] = 79;   //add new element
  arr['City'] = 'Seattle';   //change value of a key

  arr.remove('Height');   //remove an element
  arr.remove('Salary'); //remove nonexistent element
  
  print(arr.containsKey('Name'));   //search for existance of key
  print(arr.containsValue('John'));   //search for existance of value

print('\n**Modified Version**');  
  //printing by loop using function
  for (var k in arr.keys) {
      foo(k,arr);
  } 

  print('\n');
  print(arr.keys);    //printing keys
  print(arr.values);   //printing values
}